libs/auth-nest-tools/src/lib/zendeskAuth.guard.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable, CanActivate, ExecutionContext, Type } from '@nestjs/common'
+import { Request } from 'express'
+import * as crypto from 'crypto'
+
+const SIGNING_SECRET_ALGORITHM = 'sha256'
+
+export function ZendeskAuthGuard(
+  signingSecret: string | undefined,
+): Type<CanActivate> {
+  if (!signingSecret) {
+    throw new Error('Signing secret must be set')
+  }
+
+  @Injectable()
+  class ZendeskAuthGuardMixin implements CanActivate {
```

Comment:
`CanActivate` is a class, so you should be able to do the `if(!signingSecret)` check in the constructor to simplify this mixin setup.